tpm2/unseal.go

```diff
@@ -28,6 +30,64 @@ import (
 	"github.com/snapcore/secboot/internal/tcg"
 )
 
+const (
+	tryPersistentSRK = iota
+	tryTransientSRK
+	tryMax
+)
+
+func (k *SealedKeyObject) loadForUnseal(tpm *tpm2.TPMContext, session tpm2.SessionContext) (tpm2.ResourceContext, error) {
+	var lastError error
+	for try := tryPersistentSRK; try <= tryMax; try++ {
+		var srk tpm2.ResourceContext
+		if try == tryPersistentSRK {
+			var err error
+			srk, err = tpm.CreateResourceContextFromTPM(tcg.SRKHandle)
+			if tpm2.IsResourceUnavailableError(err, tcg.SRKHandle) {
+				// No SRK - save the error and try creating a transient
+				lastError = ErrTPMProvisioning
+				continue
+			} else if err != nil {
+				// This is an unexpected error
+				return nil, xerrors.Errorf("cannot create context for SRK: %w", err)
+			}
+		} else {
+			var err error
```

Comment:
is this the code that correspond to the code you mention in description that was in crypt.go?  could you add a comment here a bot the purpose of this?

Reply:
It is - I've added some documentation to this function now.